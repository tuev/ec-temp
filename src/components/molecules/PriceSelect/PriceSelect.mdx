import { Meta, Story } from '@storybook/addon-docs/blocks'
import * as stories from './PriceSelect.stories.tsx'
import PriceSelect from './PriceSelect.tsx'

<Story name="basic" />
<Story name="example" />
