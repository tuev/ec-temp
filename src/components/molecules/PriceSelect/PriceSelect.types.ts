import { SpaceProps } from 'styled-system'

export type PriceHandleChange = (
  event: React.ChangeEvent<{}>,
  value: number | number[]
) => void

export type PriceValue = [number, number]

export interface IPriceSelect {
  values?: PriceValue
  onChange?: PriceHandleChange
  min?: number
  max?: number
}

export type PriceBasicProps = SpaceProps & IPriceSelect
