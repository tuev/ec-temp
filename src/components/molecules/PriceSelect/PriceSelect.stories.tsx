import React, { useCallback, useState } from 'react'
import { PriceValue, PriceHandleChange } from './PriceSelect.types'

import PriceSelect from './PriceSelect'

export default {
  component: PriceSelect,
  includeStories: ['basic', 'example'],
  title: 'PriceSelect',
}

export const basic = () => <PriceSelect values={[10, 20]} />

export const example = () => <PriceSelectExample />

const PriceSelectExample = () => {
  const prices: PriceValue = [10, 20]
  const [values, changeValue] = useState<PriceValue>(prices)
  const handleToggleColor = useCallback(value => changeValue(value), [])

  return (
    <PriceSelect
      values={values}
      onChange={handleToggleColor}
      min={5}
      max={100}
    />
  )
}
