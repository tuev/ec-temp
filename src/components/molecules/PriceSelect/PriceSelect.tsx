import React, { FC } from 'react'
import Slider from 'components/atoms/Slider'
import { PriceBasicProps } from './PriceSelect.types'

export const defaultMethod = () => true

const PriceSelect: FC<PriceBasicProps> = (props: PriceBasicProps) => {
  const {
    values = [5, 90],
    onChange = defaultMethod,
    min = 0,
    max = 100,
  } = props

  return (
    <Slider
      showlabel={1}
      value={values}
      onChange={onChange}
      min={min}
      max={max}
    />
  )
}

export default PriceSelect
