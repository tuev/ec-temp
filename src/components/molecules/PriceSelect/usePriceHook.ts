import { useState, useCallback, useEffect } from 'react'
import { PriceValue, PriceHook } from './PriceSelect.types'

const usePriceSelect: PriceHook = (initValue = [5, 90], onEffectCallback) => {
  const [priceValue, onPriceValueChange] = useState<PriceValue>(initValue)
  const handlePriceChange = useCallback((e, value) => {
    onPriceValueChange(value)
  }, [])

  useEffect(() => {
    if (onEffectCallback && priceValue) {
      onEffectCallback(priceValue)
      handlePriceChange(null, priceValue)
    }
  }, [handlePriceChange, onEffectCallback, priceValue])

  useEffect(() => {
    handlePriceChange(null, initValue)
  }, [handlePriceChange, initValue])

  return [priceValue, handlePriceChange]
}

export default usePriceSelect
