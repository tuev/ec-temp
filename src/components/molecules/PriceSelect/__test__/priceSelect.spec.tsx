import React from 'react'

import { render, fireEvent } from '@testing-library/react'
import PriceSelect from '..'
import { defaultMethod } from '../PriceSelect'

describe('App price select', () => {
  it('Test default onChange method', () => {
    const value = defaultMethod()
    expect(value).toBe(true)
  })

  it('Test get default price select or filter no params ', () => {
    // test snapshot
    const wrapper = render(<PriceSelect />)
    expect(wrapper).toMatchSnapshot()
  })

  it('Test get default price select ', () => {
    // test snapshot
    const wrapper = render(<PriceSelect values={[10, 20]} min={5} max={100} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('Test get default price select with actions ', () => {
    const onChange = value => console.log(value, 'value')
    // test snapshot
    const wrapper = render(
      <PriceSelect values={[9, 10]} min={3} max={95} onChange={onChange} />
    )

    expect(wrapper).toMatchSnapshot()
    const { container } = wrapper
    const priceRange = container.querySelector('input')
    fireEvent.change(priceRange, { target: { value: '25,50' } })
  })
})
