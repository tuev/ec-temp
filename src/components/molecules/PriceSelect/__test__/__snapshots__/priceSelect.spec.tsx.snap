// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App price select Test get default price select  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class="MuiSlider-root MuiSlider-colorPrimary "
        showlabel="1"
      >
        <span
          class="MuiSlider-rail"
        />
        <span
          class="MuiSlider-track"
          style="left: 5.2631578947368425%; width: 10.526315789473683%;"
        />
        <input
          type="hidden"
          value="10,20"
        />
        <span
          aria-orientation="horizontal"
          aria-valuemax="100"
          aria-valuemin="5"
          aria-valuenow="10"
          aria-valuetext="10"
          class="MuiSlider-thumb MuiSlider-thumbColorPrimary PrivateValueLabel-thumb-46"
          data-index="0"
          role="slider"
          style="left: 5.2631578947368425%;"
          tabindex="0"
        >
          <span
            class="PrivateValueLabel-offset-48 MuiSlider-valueLabel"
          >
            <span
              class="PrivateValueLabel-circle-49"
            >
              <span
                class="PrivateValueLabel-label-50"
              >
                10
              </span>
            </span>
          </span>
        </span>
        <span
          aria-orientation="horizontal"
          aria-valuemax="100"
          aria-valuemin="5"
          aria-valuenow="20"
          aria-valuetext="20"
          class="MuiSlider-thumb MuiSlider-thumbColorPrimary PrivateValueLabel-thumb-46"
          data-index="1"
          role="slider"
          style="left: 15.789473684210526%;"
          tabindex="0"
        >
          <span
            class="PrivateValueLabel-offset-48 MuiSlider-valueLabel"
          >
            <span
              class="PrivateValueLabel-circle-49"
            >
              <span
                class="PrivateValueLabel-label-50"
              >
                20
              </span>
            </span>
          </span>
        </span>
      </span>
      <div
        style="display: flex; justify-content: space-between;"
      >
        <span>
          10
        </span>
        <span>
          20
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <span
      class="MuiSlider-root MuiSlider-colorPrimary sc-AykKC KUWBx"
      showlabel="1"
    >
      <span
        class="MuiSlider-rail"
      />
      <span
        class="MuiSlider-track"
        style="left: 5.2631578947368425%; width: 10.526315789473683%;"
      />
      <input
        type="hidden"
        value="10,20"
      />
      <span
        aria-orientation="horizontal"
        aria-valuemax="100"
        aria-valuemin="5"
        aria-valuenow="10"
        aria-valuetext="10"
        class="MuiSlider-thumb MuiSlider-thumbColorPrimary PrivateValueLabel-thumb-46"
        data-index="0"
        role="slider"
        style="left: 5.2631578947368425%;"
        tabindex="0"
      >
        <span
          class="PrivateValueLabel-offset-48 MuiSlider-valueLabel"
        >
          <span
            class="PrivateValueLabel-circle-49"
          >
            <span
              class="PrivateValueLabel-label-50"
            >
              10
            </span>
          </span>
        </span>
      </span>
      <span
        aria-orientation="horizontal"
        aria-valuemax="100"
        aria-valuemin="5"
        aria-valuenow="20"
        aria-valuetext="20"
        class="MuiSlider-thumb MuiSlider-thumbColorPrimary PrivateValueLabel-thumb-46"
        data-index="1"
        role="slider"
        style="left: 15.789473684210526%;"
        tabindex="0"
      >
        <span
          class="PrivateValueLabel-offset-48 MuiSlider-valueLabel"
        >
          <span
            class="PrivateValueLabel-circle-49"
          >
            <span
              class="PrivateValueLabel-label-50"
            >
              20
            </span>
          </span>
        </span>
      </span>
    </span>
    <div
      style="display: flex; justify-content: space-between;"
    >
      <span>
        10
      </span>
      <span>
        20
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`App price select Test get default price select or filter no params  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class="MuiSlider-root MuiSlider-colorPrimary "
        showlabel="1"
      >
        <span
          class="MuiSlider-rail"
        />
        <span
          class="MuiSlider-track"
          style="left: 5%; width: 85%;"
        />
        <input
          type="hidden"
          value="5,90"
        />
        <span
          aria-orientation="horizontal"
          aria-valuemax="100"
          aria-valuemin="0"
          aria-valuenow="5"
          aria-valuetext="5"
          class="MuiSlider-thumb MuiSlider-thumbColorPrimary PrivateValueLabel-thumb-21"
          data-index="0"
          role="slider"
          style="left: 5%;"
          tabindex="0"
        >
          <span
            class="PrivateValueLabel-offset-23 MuiSlider-valueLabel"
          >
            <span
              class="PrivateValueLabel-circle-24"
            >
              <span
                class="PrivateValueLabel-label-25"
              >
                5
              </span>
            </span>
          </span>
        </span>
        <span
          aria-orientation="horizontal"
          aria-valuemax="100"
          aria-valuemin="0"
          aria-valuenow="90"
          aria-valuetext="90"
          class="MuiSlider-thumb MuiSlider-thumbColorPrimary PrivateValueLabel-thumb-21"
          data-index="1"
          role="slider"
          style="left: 90%;"
          tabindex="0"
        >
          <span
            class="PrivateValueLabel-offset-23 MuiSlider-valueLabel"
          >
            <span
              class="PrivateValueLabel-circle-24"
            >
              <span
                class="PrivateValueLabel-label-25"
              >
                90
              </span>
            </span>
          </span>
        </span>
      </span>
      <div
        style="display: flex; justify-content: space-between;"
      >
        <span>
          5
        </span>
        <span>
          90
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <span
      class="MuiSlider-root MuiSlider-colorPrimary sc-AykKC KUWBx"
      showlabel="1"
    >
      <span
        class="MuiSlider-rail"
      />
      <span
        class="MuiSlider-track"
        style="left: 5%; width: 85%;"
      />
      <input
        type="hidden"
        value="5,90"
      />
      <span
        aria-orientation="horizontal"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="5"
        aria-valuetext="5"
        class="MuiSlider-thumb MuiSlider-thumbColorPrimary PrivateValueLabel-thumb-21"
        data-index="0"
        role="slider"
        style="left: 5%;"
        tabindex="0"
      >
        <span
          class="PrivateValueLabel-offset-23 MuiSlider-valueLabel"
        >
          <span
            class="PrivateValueLabel-circle-24"
          >
            <span
              class="PrivateValueLabel-label-25"
            >
              5
            </span>
          </span>
        </span>
      </span>
      <span
        aria-orientation="horizontal"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="90"
        aria-valuetext="90"
        class="MuiSlider-thumb MuiSlider-thumbColorPrimary PrivateValueLabel-thumb-21"
        data-index="1"
        role="slider"
        style="left: 90%;"
        tabindex="0"
      >
        <span
          class="PrivateValueLabel-offset-23 MuiSlider-valueLabel"
        >
          <span
            class="PrivateValueLabel-circle-24"
          >
            <span
              class="PrivateValueLabel-label-25"
            >
              90
            </span>
          </span>
        </span>
      </span>
    </span>
    <div
      style="display: flex; justify-content: space-between;"
    >
      <span>
        5
      </span>
      <span>
        90
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`App price select Test get default price select with actions  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class="MuiSlider-root MuiSlider-colorPrimary "
        showlabel="1"
      >
        <span
          class="MuiSlider-rail"
        />
        <span
          class="MuiSlider-track"
          style="left: 6.521739130434782%; width: 1.0869565217391308%;"
        />
        <input
          type="hidden"
          value="9,10"
        />
        <span
          aria-orientation="horizontal"
          aria-valuemax="95"
          aria-valuemin="3"
          aria-valuenow="9"
          aria-valuetext="9"
          class="MuiSlider-thumb MuiSlider-thumbColorPrimary PrivateValueLabel-thumb-71"
          data-index="0"
          role="slider"
          style="left: 6.521739130434782%;"
          tabindex="0"
        >
          <span
            class="PrivateValueLabel-offset-73 MuiSlider-valueLabel"
          >
            <span
              class="PrivateValueLabel-circle-74"
            >
              <span
                class="PrivateValueLabel-label-75"
              >
                9
              </span>
            </span>
          </span>
        </span>
        <span
          aria-orientation="horizontal"
          aria-valuemax="95"
          aria-valuemin="3"
          aria-valuenow="10"
          aria-valuetext="10"
          class="MuiSlider-thumb MuiSlider-thumbColorPrimary PrivateValueLabel-thumb-71"
          data-index="1"
          role="slider"
          style="left: 7.608695652173913%;"
          tabindex="0"
        >
          <span
            class="PrivateValueLabel-offset-73 MuiSlider-valueLabel"
          >
            <span
              class="PrivateValueLabel-circle-74"
            >
              <span
                class="PrivateValueLabel-label-75"
              >
                10
              </span>
            </span>
          </span>
        </span>
      </span>
      <div
        style="display: flex; justify-content: space-between;"
      >
        <span>
          9
        </span>
        <span>
          10
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <span
      class="MuiSlider-root MuiSlider-colorPrimary sc-AykKC KUWBx"
      showlabel="1"
    >
      <span
        class="MuiSlider-rail"
      />
      <span
        class="MuiSlider-track"
        style="left: 6.521739130434782%; width: 1.0869565217391308%;"
      />
      <input
        type="hidden"
        value="9,10"
      />
      <span
        aria-orientation="horizontal"
        aria-valuemax="95"
        aria-valuemin="3"
        aria-valuenow="9"
        aria-valuetext="9"
        class="MuiSlider-thumb MuiSlider-thumbColorPrimary PrivateValueLabel-thumb-71"
        data-index="0"
        role="slider"
        style="left: 6.521739130434782%;"
        tabindex="0"
      >
        <span
          class="PrivateValueLabel-offset-73 MuiSlider-valueLabel"
        >
          <span
            class="PrivateValueLabel-circle-74"
          >
            <span
              class="PrivateValueLabel-label-75"
            >
              9
            </span>
          </span>
        </span>
      </span>
      <span
        aria-orientation="horizontal"
        aria-valuemax="95"
        aria-valuemin="3"
        aria-valuenow="10"
        aria-valuetext="10"
        class="MuiSlider-thumb MuiSlider-thumbColorPrimary PrivateValueLabel-thumb-71"
        data-index="1"
        role="slider"
        style="left: 7.608695652173913%;"
        tabindex="0"
      >
        <span
          class="PrivateValueLabel-offset-73 MuiSlider-valueLabel"
        >
          <span
            class="PrivateValueLabel-circle-74"
          >
            <span
              class="PrivateValueLabel-label-75"
            >
              10
            </span>
          </span>
        </span>
      </span>
    </span>
    <div
      style="display: flex; justify-content: space-between;"
    >
      <span>
        9
      </span>
      <span>
        10
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
